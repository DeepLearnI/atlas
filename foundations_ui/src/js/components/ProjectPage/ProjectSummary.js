import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import HoverCell from '../JobListPage/cells/HoverCell';
import CommonActions from '../../actions/CommonActions';

const ProjectSummary = props => {
  const [showAllTags, setShowAllTags] = React.useState(false);

  const packageClick = () => {
    const { history, project } = props;

    history.push(
      `/projects/${project.name}/job_listing`,
      {
        project: project,
      },
    );
  };

  const onClickShowTags = () => {
    const { project } = props;
    const value = !showAllTags;
    setShowAllTags(value);
  };

  const onMouseLeave = () => {
    setShowAllTags(false);
  };

  const { project } = props;

  const expandedTagSpans = [];
  project.tags.forEach(tag => {
    expandedTagSpans.push(<span key={'tag-'.concat(tag)}>{tag}</span>);
  });

  return (
    <div
      className="project-summary-container"
      data-class="project-summary"
    >
      <div className="project-summary-info-container">
        <h2
          onClick={packageClick}
          // eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
          role="button"
          tabIndex={0}
          onKeyPress={packageClick}
          className="font-bold"
        >
          {project.name}
        </h2>
        <p className="font-bold">
          Created at: <span>{CommonActions.formatDate(project.created_at)}</span>
        </p>
        <div className="project-summary-button-container" />
      </div>
      <div className="project-summary-tags-container">
        <p>tags</p>
        {project.tags.slice(0, 10).map(tag => {
          return <span key={tag}>{tag}</span>;
        })}
        {project.tags.length > 10
          && (
            <span
              className="span-more"
              role="presentation"
              onClick={onClickShowTags}
              onKeyDown={() => {}}
            >
              ...
            </span>
          )
        }
        {showAllTags && <HoverCell onMouseLeave={onMouseLeave} textToRender={expandedTagSpans} />}
      </div>
    </div>
  );
};

ProjectSummary.propTypes = {
  project: PropTypes.object,
  history: PropTypes.object,
};

ProjectSummary.defaultProps = {
  project: {},
  history: {},
};

export default withRouter(ProjectSummary);
