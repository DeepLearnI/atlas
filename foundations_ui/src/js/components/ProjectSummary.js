import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProjectSummary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      project: this.props.project,
    };
  }

  render() {
    const { project } = this.state;
    return (
      <div className="project-summary-container elevation-1">
        <div className="project-summary-info-container">
          <h2 className="project-summary-name-text">{project.name}</h2>
          <p className="project-summary-source-text">data source</p>
          <p className="project-summary-owner-text">
            Project owner: <span className="font-regular">{project.owner}</span>
          </p>
          <p className="project-summary-created-at-text">
            Created at: <span className="font-regular">{project.created_at}</span>
          </p>
          <div className="project-summary-button-container">
            <button type="button" className="b--mat b--affirmative">view queue</button>
            <button type="button" className="b--mat b--affirmative">view job list</button>
          </div>
        </div>
        <div className="project-summary-metrics-container" />
      </div>
    );
  }
}

ProjectSummary.propTypes = {
  project: PropTypes.object,
};

ProjectSummary.defaultProps = {
  project: {},
};

export default ProjectSummary;
