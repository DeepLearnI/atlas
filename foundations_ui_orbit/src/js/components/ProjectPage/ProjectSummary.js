import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";

const ProjectSummary = props => {
  const packageClick = () => {
    const { history, project } = props;

    history.push(
      `/projects/${project.name}/schedules`,
      {
        project: project
      }
    );
  };

  const { project } = props;


  return (
    <div
      className="project-summary-container elevation-1"
      onClick={packageClick}
    >
      <div className="project-summary-info-container">
        <h2 className="font-bold">{project.name}</h2>
        <p>Data Source: Unknown</p>
        <p className="font-bold">
          Project owner: <span>{project.owner}</span>
        </p>
        <p className="font-bold">
          Created at: <span>{project.created_at}</span>
        </p>
        <div className="project-summary-button-container">
          {/* <button type="button" className="b--mat b--affirmative">
              view queue
            </button>
            <button type="button" className="b--mat b--affirmative">
              view job list
            </button> */}
          {/* <button
              type="button"
              onClick={this.packageClick}
              className="b--mat b--affirmative"
            >
              Post Deploy Management
            </button> */}
        </div>
      </div>
      <div className="project-summary-metrics-container" />
    </div>
  );
};

ProjectSummary.propTypes = {
  project: PropTypes.object,
  history: PropTypes.object
};

ProjectSummary.defaultProps = {
  project: {},
  history: {}
};

export default withRouter(ProjectSummary);
