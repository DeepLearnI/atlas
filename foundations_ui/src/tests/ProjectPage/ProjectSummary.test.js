import React from 'react';
import ReactDOM from 'react-dom';
import ProjectSummary from '../../js/components/ProjectPage/ProjectSummary';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

/*it('Shallow Renders ProjectSummary', () => {
  const wrapper = shallow(<ProjectSummary/>);
  expect(wrapper).toMatchSnapshot();
});*/

/*it('Has Projects', () => {
  const project = {
      name: "test",
      owner: "me",
      created_at: "today",
  };
  const wrapper = mount(<ProjectSummary project={project}/>); 
  const state = wrapper.state();
  expect(state.project.name.length > 0);
  expect(state.project.owner.length > 0);
  expect(state.project.created_at.length > 0);
});

it('Sets Required State On Button Click', () => {
  const project = {
    name: "test",
    owner: "me",
    created_at: "today",
  };
  const projectWrapper = shallow(<ProjectSummary project={project}/>);
  ProjectActions.redirect = jest.fn();
  projectWrapper.find('div').at(0).simulate('click');
  expect(projectWrapper.state('redirect')).toEqual(true);
})

it('Sends Correct Path To Redirect', () => {
  ProjectActions.redirect = jest.fn();
  const project = {
    name: "test",
    owner: "me",
    created_at: "today",
  };
  const projectWrapper = shallow(<ProjectSummary project={project}/>);
  expect(ProjectActions.redirect).not.toHaveBeenCalled();
  projectWrapper.setState({ redirect: true});
  expect(ProjectActions.redirect).toHaveBeenCalledWith("/projects/test/job_listing");
});*/

it('needs a dummy test to pass', () => {

});

