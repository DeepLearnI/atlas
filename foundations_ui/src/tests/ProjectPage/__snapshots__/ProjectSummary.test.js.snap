// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders ProjectSummary 1`] = `
<div
  className="project-summary-container elevation-1"
>
  <div
    className="project-summary-info-container"
  >
    <h2
      className="font-bold"
    />
    <p
      className="font-bold"
    >
      Data Source:
      <span>
         raw_mnist
      </span>
    </p>
    <p
      className="font-bold"
    >
      Project owner: 
      <span>
        Trial
      </span>
    </p>
    <p
      className="font-bold"
    >
      Created at: 
      <span>
        2019-03-30 T 10:45 UTC
      </span>
    </p>
    <div
      className="project-summary-button-container"
    >
      <button
        className="b--mat b--affirmative"
        onClick={[Function]}
        type="button"
      >
        View jobs
      </button>
    </div>
  </div>
  <div
    className="project-summary-metrics-container"
  />
</div>
`;
